unleash-yggdrasil/src/lib.rs

```diff
@@ -622,6 +618,26 @@ pub struct VariantDef {
     pub enabled: bool,
 }
 
+impl VariantDef {
+    pub fn to_enriched_response(&self, toggle_enabled: bool) -> ExtendedVariantDef {
+        ExtendedVariantDef {
+            name: self.name.clone(),
+            payload: self.payload.clone(),
+            enabled: self.enabled,
+            feature_enabled: toggle_enabled,
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq, Clone)]
+pub struct ExtendedVariantDef {
```

Comment:
Honestly not loving this name. If anyone has a better one please feel free to suggest

Reply:
Same. Honestly I see this simply as a VariantDef, since this is how our variants now look like, with the `feature_enabled` property as part of them. So we're saying a variant is now effectively:

```
{
  name,
  enabled,
  payload,
  feature_enabled
}
```

Which means your `variant` variable is not really a variant, and `check_variant_by_toggle` doesn't really return a variant, but rather parts of the variant.

Perhaps something like:
 - `VariantDef` -> `BaseVariantDef`
 - `ExtendedVariantDef` -> `VariantDef`
 
 But to be honest I'm not 100% sure. I think this needs more input.